storage/timecache/timeCacher.go

```diff
@@ -115,14 +115,15 @@ func (tc *timeCacher) Peek(key []byte) (value interface{}, ok bool) {
 // HasOrAdd checks if a key is in the cache.
 // If key exists, does not update the value. Otherwise, adds the key-value in the cache
 func (tc *timeCacher) HasOrAdd(key []byte, value interface{}, _ int) (has, added bool) {
-	existed, err := tc.timeCache.upsert(string(key), value, tc.timeCache.defaultSpan)
+	var err error
+	has, added, err = tc.timeCache.hasOrAdd(string(key), value, tc.timeCache.defaultSpan)
```

Comment:
`has` and `added` are mutually exclusive, right? Bit of redundancy therefore, but it's good to be explicit :+1:

Reply:
yes, they are exclusive. Since I have to implement the already existing and used interface, I had to return something.
Also, beside (true, false) and (false, true) we also have (false, false) when the provided key is nil or empty, so we had to encode 3 states.